src/main/java/com/spotify/docker/client/SystemCredentialHelperDelegate.java

```diff
@@ -0,0 +1,139 @@
+/*-
+ * -\-\-
+ * docker-client
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.docker.client;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.spotify.docker.client.DockerCredentialHelper.CredentialHelperDelegate;
+import com.spotify.docker.client.messages.DockerCredentialHelperAuth;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The default credential helper delegate.
+ * Executes each credential helper operation on the system.
+ */
+@SuppressFBWarnings("SIC_INNER_SHOULD_BE_STATIC_ANON")
+class SystemCredentialHelperDelegate implements CredentialHelperDelegate {
+
+  private static final Logger log = LoggerFactory.getLogger(DockerConfigReader.class);
+  private static final ObjectMapper mapper = ObjectMapperProvider.objectMapper();
+
+  @Override
+  public int store(final String credsStore, final DockerCredentialHelperAuth auth)
+      throws IOException, InterruptedException {
+    final Process process = exec("store", credsStore);
+
+    try (final Writer outStreamWriter =
+             new OutputStreamWriter(process.getOutputStream(), StandardCharsets.UTF_8)) {
+      try (final BufferedWriter writer = new BufferedWriter(outStreamWriter)) {
+        writer.write(mapper.writeValueAsString(auth) + "\n");
```

Comment:
I didn't notice this in the prior version of the code, but I wonder if all of these `"\n"` strings should instead be `System.lineSeparator()` to avoid assuming the code only runs on *nux systems.

Reply:
good point. I'll change.